nautobot/extras/jobs.py

```diff
@@ -815,6 +837,7 @@ def run_job(data, request, job_result_pk, commit=True, *args, **kwargs):
         return False
     job = job_class()
     job.job_result = job_result
+    data = job_class.deserialize_data(data)
```

Comment:
If deserialization fails (e.g. due to a record `DoesNotExist` exception as mentioned above) we need to handle that gracefully, mark the job as `STATUS_ERRORED`, etc.)